toonz/sources/toonzlib/txshsimplelevel.cpp

```diff
@@ -1374,8 +1374,26 @@ void TXshSimpleLevel::save() {
 //-----------------------------------------------------------------------------
 
 static void saveBackup(TFilePath path) {
+  int totalBackups = Preferences::instance()->getBackupKeepCount();
+  totalBackups -= 1;
+  TFilePath backup = path.withType(path.getType() + ".bak");
+  TFilePath prevBackup =
+      path.withType(path.getType() + ".bak" + std::to_string(totalBackups));
+  while (--totalBackups >= 0) {
+    std::string bakExt =
+        ".bak" + (totalBackups > 0 ? std::to_string(totalBackups) : "");
+    backup = path.withType(path.getType() + bakExt);
+    if (!TSystem::doesExistFileOrLevel(backup)) continue;
+    try {
+      if (TSystem::doesExistFileOrLevel(prevBackup))
+        TSystem::removeFileOrLevel_throw(prevBackup);
+      TSystem::copyFileOrLevel_throw(prevBackup, backup);
+    } catch (...) {
+    }
+    prevBackup = backup;
+  }
+
   try {
-    TFilePath backup = path.withName(path.getName() + "_backup");
     if (TSystem::doesExistFileOrLevel(backup))
       TSystem::removeFileOrLevel_throw(backup);
     TSystem::copyFileOrLevel_throw(backup, path);
```

Comment:
Currently it fails to save the backup properly when you save the raster levels (like `A..tif` ).

Since the backup extension has changed to `.bak` , OT seems to fail copying the level here as the target level name becomes with the single period ( like `A..tif.bak` ), which is recognized as single image.

Reply:
The individual frames are still saving but on new levels being saved for the 1st time, it creates an odd file because of the sequence range.

This appears to be an existing bug. In the current version, I observed the files would be saved as A_backup.0001.tif, A.0001.tif, A.0002.tif, etc. If you delete A_backup.0001.tif and make changes to the frames then save, A_backup.0001.tif does not get created.

I will fix this bug with this release.

Reply:
I just discovered why this does not work for sequenced files. The change in actual extension causes OT to no longer acknowledge jt as sequenced files.  I'll need to figure out the best way to deal with this.